org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeBinding.java

```diff
@@ -702,8 +704,7 @@ protected String getQualifiedNameImpl(Type type, TypeSymbol typeSymbol, Symbol o
 
 		if( includeParameters ) {
 			ITypeBinding[] typeArguments = getUncheckedTypeArguments(type, typeSymbol);
-			boolean isTypeDeclaration = this.isDeclaration; //typeSymbol != null && typeSymbol.type == type;
-			if (!isTypeDeclaration && typeArguments.length > 0) {
+			if (this.specificInstance && typeArguments.length > 0) {
```

Comment:
I think the issue here is simply that we should use `getTypeParameters` instead of `getTypeArguments`. getTypeParameters must return an empty array for generics.